docs/source/app-dev/bindings-x-lang/index.rst

```diff
@@ -0,0 +1,92 @@
+    Code generator. A program that generates classes representing DAML contract templates in X language. These classes incorporate all boilerplate code for constructing: :ref:`com.digitalasset.ledger.api.v1.CreateCommand` and :ref:`com.digitalasset.ledger.api.v1.ExerciseCommand` corresponding for each DAML contract template.
+
+Technically codegen is optional. It is possible to construct the commands manually. However it is very tedious and error-prone.
+
+Building Create Command
+=======================
+
+Let's recall an **IOU** example from the :doc:`Quickstart guide </getting-started/quickstart>`, where `Iou` template is defined like this:
+
+.. literalinclude:: ./code-snippets/quickstart/template-root/daml/Iou.daml
+  :language: daml
+  :lines: 9-15
+
+Here is how to manually build :ref:`com.digitalasset.ledger.api.v1.CreateCommand` for the above contract template in Scala:
+
+.. literalinclude:: ./code-snippets/iou-no-codegen/application/src/main/scala/com/digitalasset/quickstart/iou/IouCommands.scala
+   :start-after: // <doc-ref:iou-no-codegen-create-command>
+   :end-before: // </doc-ref:iou-no-codegen-create-command>
+
+If you do not specify any of the above fields or type their names or values incorrectly, or do not order them exactly as they are in the DAML template, the above code will compile but fail at run-time because you did not structure your create command correctly. Codegen is expected to simplify the command construction by providing auto-generated utilities to help you construct commands.
+
+When you use :doc:`Scala codegen </app-dev/bindings-scala/index>` to generate contract classes, a similar contract instantiation would look:
+
+.. literalinclude:: ./code-snippets/quickstart-scala/application/src/main/scala/com/digitalasset/quickstart/iou/IouMain.scala
+   :start-after: // <doc-ref:iou-contract-instance>
+   :end-before: // </doc-ref:iou-contract-instance>
+
+Building Exercise Command
+=========================
+
+To build :ref:`com.digitalasset.ledger.api.v1.ExerciseCommand` for `Iou_Transfer`:
+
+.. literalinclude:: ./code-snippets/quickstart/template-root/daml/Iou.daml
+  :language: daml
+  :lines: 23, 52-55
+
+manually in Scala:
+
+.. literalinclude:: ./code-snippets/iou-no-codegen/application/src/main/scala/com/digitalasset/quickstart/iou/IouCommands.scala
+   :start-after: // <doc-ref:iou-no-codegen-exercise-command>
+   :end-before: // </doc-ref:iou-no-codegen-exercise-command>
+
+versus creating the same command using a value class generated by :doc:`Scala codegen </app-dev/bindings-scala/index>`:
+
+.. literalinclude:: ./code-snippets/quickstart-scala/application/src/main/scala/com/digitalasset/quickstart/iou/IouMain.scala
+   :start-after: // <doc-ref:iou-exercise-transfer-cmd>
+   :end-before: // </doc-ref:iou-exercise-transfer-cmd>
+
+Summary
+=======
+
+If you are creating custom bindings for the Digital Asset distributed ledger, you will need to:
+
+- generate **Ledger API** from the gRPC definitions
```

Comment:
This is a little bit light on detail. Is there anything we can link to in the gRPC docs to explain how to do this? Otherwise it's a bit ["draw the rest of the owl"](https://www.google.com/search?q=draw+the+owl&source=lnms&tbm=isch&sa=X&ved=0ahUKEwiC5ZDviOHiAhXTwsQBHflmAE0Q_AUIECgB&biw=1280&bih=721#imgrc=gbPGHaXIjOhvjM:).

Reply:
it is just a summary to re-iterate what needs to be done. The **Ledger API** doc and the link to gRPC tutorial above cover this up.

Reply:
I am also fine with removing this completely.